Code/EnginePlugins/RmlUiPlugin/Resources/RmlUiResource.cpp

```diff
@@ -0,0 +1,151 @@
+
+  stream.WriteVersion(s_RmlUiDescVersion);
+
+  stream << m_sRmlFile;
+  stream << m_ScaleMode;
+  stream << m_ReferenceResolution;
+
+  return EZ_SUCCESS;
+}
+
+ezResult ezRmlUiResourceDescriptor::Load(ezStreamReader& stream)
+{
+  m_DependencyFile.ReadDependencyFile(stream);
+
+  ezTypeVersion uiVersion = stream.ReadVersion(s_RmlUiDescVersion);
+
+  stream >> m_sRmlFile;
+  stream >> m_ScaleMode;
+  stream >> m_ReferenceResolution;
+
+  return EZ_SUCCESS;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+// clang-format off
+EZ_BEGIN_DYNAMIC_REFLECTED_TYPE(ezRmlUiResource, 1, ezRTTIDefaultAllocator<ezRmlUiResource>)
+EZ_END_DYNAMIC_REFLECTED_TYPE;
+
+EZ_RESOURCE_IMPLEMENT_COMMON_CODE(ezRmlUiResource);
+// clang-format on
+
+ezRmlUiResource::ezRmlUiResource()
+  : ezResource(DoUpdate::OnAnyThread, 1)
+{
+}
+
+ezResourceLoadDesc ezRmlUiResource::UnloadData(Unload WhatToUnload)
+{
+  ezResourceLoadDesc res;
+  res.m_uiQualityLevelsDiscardable = 0;
+  res.m_uiQualityLevelsLoadable = 0;
+  res.m_State = ezResourceState::Unloaded;
+
+  return res;
+}
+
+ezResourceLoadDesc ezRmlUiResource::UpdateContent(ezStreamReader* Stream)
+{
+  ezRmlUiResourceDescriptor desc;
+  ezResourceLoadDesc res;
+  res.m_uiQualityLevelsDiscardable = 0;
+  res.m_uiQualityLevelsLoadable = 0;
+
+  if (Stream == nullptr)
+  {
+    res.m_State = ezResourceState::LoadedResourceMissing;
+    return res;
+  }
+
+  ezStringBuilder sAbsFilePath;
+  (*Stream) >> sAbsFilePath;
+
+  // Direct loading of rml file
+  if (sAbsFilePath.GetFileExtension() == "rml")
+  {
+    m_sRmlFile = sAbsFilePath;
+
+    res.m_State = ezResourceState::Loaded;
+    return res;
+  }
+
+  ezAssetFileHeader assetHeader;
+  assetHeader.Read(*Stream);
+
+  if (desc.Load(*Stream).Failed())
+  {
+    res.m_State = ezResourceState::LoadedResourceMissing;
+    return res;
+  }
+
+  return CreateResource(std::move(desc));
+}
+
+void ezRmlUiResource::UpdateMemoryUsage(MemoryUsage& out_NewMemoryUsage)
+{
+  // TODO
+  out_NewMemoryUsage.m_uiMemoryCPU = sizeof(*this);
+  out_NewMemoryUsage.m_uiMemoryGPU = 0;
+}
+
+EZ_RESOURCE_IMPLEMENT_CREATEABLE(ezRmlUiResource, ezRmlUiResourceDescriptor)
+{
+  m_sRmlFile = descriptor.m_sRmlFile;
+  m_ScaleMode = descriptor.m_ScaleMode;
+  m_ReferenceResolution = descriptor.m_ReferenceResolution;
+
+  ezResourceLoadDesc res;
+  res.m_uiQualityLevelsDiscardable = 0;
+  res.m_uiQualityLevelsLoadable = 0;
+  res.m_State = ezResourceState::Loaded;
+
+  return res;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+bool ezRmlUiResourceLoader::IsResourceOutdated(const ezResource* pResource) const
+{
+  if (ezResourceLoaderFromFile::IsResourceOutdated(pResource))
+    return true;
+
+  ezStringBuilder sId = pResource->GetResourceID();
+  if (sId.GetFileExtension() == "rml")
+    return false;
+
+  ezFileReader stream;
+  if (stream.Open(pResource->GetResourceID()).Failed())
```

Comment:
Not sure if it's a good idea to mark it as outdated, when you can't read the file.